homeassistant/components/command_line/binary_sensor.py

```diff
@@ -59,16 +59,30 @@ async def async_setup_platform(
 ) -> None:
     """Set up the Command line Binary Sensor."""
 
-    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
-
-    name: str = config.get(CONF_NAME, DEFAULT_NAME)
-    command: str = config[CONF_COMMAND]
-    payload_off: str = config[CONF_PAYLOAD_OFF]
-    payload_on: str = config[CONF_PAYLOAD_ON]
-    device_class: BinarySensorDeviceClass | None = config.get(CONF_DEVICE_CLASS)
-    value_template: Template | None = config.get(CONF_VALUE_TEMPLATE)
-    command_timeout: int = config[CONF_COMMAND_TIMEOUT]
-    unique_id: str | None = config.get(CONF_UNIQUE_ID)
+    if binary_sensor_config := config:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "deprecated_yaml_binary_sensor",
+            breaks_in_ha_version="2023.8.0",
+            is_fixable=False,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_platform_yaml",
+            translation_placeholders={"platform": "Binary Sensor"},
```

Comment:
This won't be translated, I suggest to just use the binary sensor domain + quoting it in the description.
```suggestion
from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
            translation_placeholders={"platform": BINARY_SENSOR_DOMAIN},
```